spec/factories/opponents.rb

```diff
@@ -18,6 +19,17 @@
         opponent.opposable.save! if evaluator.first_name || evaluator.last_name
       end
     end
+
+    trait :for_organisation do
+      opposable factory: :organisation
+
+      after(:create) do |opponent, evaluator|
+        opponent.opposable.name = evaluator.organisation_name if evaluator.organisation_name
+        opponent.opposable.ccms_code = evaluator.organisation_ccms_code if evaluator.organisation_ccms_code
+        opponent.opposable.description = evaluator.organisation_description if evaluator.organisation_description
+        opponent.opposable.save! if evaluator.organisation_name || evaluator.organisation_ccms_code || evaluator.organisation_description
+      end
```

Comment:
Feel convoluted, have you considered refactoring?

Reply:
how would you refactor?

Reply:
not many options which would be more pleasant TBH, as discussed